CONTRIBUTING.md

```diff
@@ -0,0 +1,300 @@
+
+* A short descriptive title
+* Number
+* Type
+* Status
+* The GitHub usernames of the author(s)
+* The GitHub username of the owner
+* The date the PEP was created
+
+
+### Abstract
+
+A short (~200 word) description of the technical issue being addressed.
+
+
+### Motivation
+
+It should clearly explain why Porter's existing functionality is inadequate to
+address the problem that the PEP solves and identify the impacted audience(s) (mixin
+developers, bundle authors, end-users). PEP submissions without sufficient
+motivation may be rejected outright. This is the most important part at the
+beginning and is required before moving forward.
+
+
+### Rationale
+
+The rationale fleshes out the specification by describing why particular design
+decisions were made.  It should describe alternate designs that were considered
+and related work.
+
+The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
+
+
+### Specification
+
+The technical specification should describe the command and/or configuration
+syntax and semantics of any new feature.
+
+* If this is a command, we are looking for what the `porter help` would look
+  like: description of command, arguments, flags, default behavior and error
+  handling.
+
+* If this is a syntax change to a configuration file, define the allowed syntax,
+  at least one example per use case, covering defaults and error handling.
+
+* All PEPs will be reviewed for user experience. So make sure to think about the
+  common use case, how people can accomplish more advanced scenarios, precedence
+  from existing Porter features or other tools in the ecosystem, and how the
+  change fits into Porter workflows and tasks.
+
+
+### Backwards Compatibility
+
+All PEPs that introduce backwards incompatibilities must include a section
+describing these incompatibilities and their severity.  The PEP must explain how
+to deal with these incompatibilities, possibly with defaulting or migrations.
+PEP submissions without a sufficient backwards compatibility treatise may be
+rejected outright.
+
+
+### Security Implications
+
+If there are security concerns in relation to the PEP, those concerns should be
+explicitly written out to make sure reviewers of the PEP are aware of them.
+
+
+### Rejected Ideas
+
+Throughout the discussion of a PEP, various ideas will be proposed which are not
+accepted. Those rejected ideas should be recorded along with the reasoning as to
+why they were rejected. This both helps record the thought process behind the
+final version of the PEP as well as preventing people from bringing up the same
+rejected idea again in subsequent discussions.
+
+
+### Open Questions
+
+While a PEP is in draft, questions can come up which warrant further discussion.
+Those questions should be recorded here so people know that they are being
+thought about but do not have a concrete resolution. This helps make sure all
```

Comment:
Missing word here?  Currently "This helps make sure all (?) for the PEP to be ready..."